float8_experimental/float8_dynamic_linear.py

```diff
@@ -96,8 +111,110 @@ def from_float(cls, mod, emulate: bool = False) -> "Float8DynamicLinear":
                 "bias": False,
             }
             new_mod = cls(**super_kwargs)
-        new_mod.weight = mod.weight
-        new_mod.bias = mod.bias
         new_mod.forward_config = ScaledMMConfig(emulate, True if not emulate else False)
         new_mod.backward_config = ScaledMMConfig(emulate, False)
+        if config.enable_fsdp_fp8_all_gather:
+            new_mod.weight = nn.Parameter(
+                WeightWithDynamicFloat8CastTensor(mod.weight, new_mod.forward_config)
```

Comment:
Following the existing behavior without fp8 all-gather, we always use the `forward_config` for the weight.

Reply:
yeah looks right